src/shogun/classifier/svm/SVM.h

```diff
@@ -242,6 +242,21 @@ class CSVM : public CKernelMachine
 		void set_callback_function(CMKL* m, bool (*cb)
 				(CMKL* mkl, const float64_t* sumw, const float64_t suma));
 
+		/** @return whether mkl interleaving optimization is possible or not */
+		virtual bool supports_interleaving() const { return false; }
+
+		/** @return whether this svm is one-class or multi-class */
+		virtual bool is_one_class() const { return false; }
+
+		/** @return whether this svm is multiclass or one-class*/
+		bool is_multiclass() const { return !is_one_class(); }
```

Comment:
why this?

Reply:
I was thinking it maybe should be used in MKLMulticlass?

Reply:
no that would be quite different ...

Reply:
As I understand it, the one class SVMs, are used with the MKLOneClass, and similarily MKLMulticlass is used with the rest, so MKLMulticlass should check for this condition. If that's not the case, the it's useless

Reply:
This can be replaced by EProblemType as well, since it has unary, binary, and multiclass.